src/GraphQL/Resolver/Element.php

```diff
@@ -136,6 +148,165 @@ public function resolveSiblings($value = null, $args = [], $context, ResolveInfo
+        if (isset($args['ids'])) {
+            $conditionParts[] = '(o_id IN (' . $args['ids'] . '))';
+        }
+
+        // paging
+        if (isset($args['first'])) {
+            $objectList->setLimit($args['first']);
+        }
+
+        if (isset($args['after'])) {
+            $objectList->setOffset($args['after']);
+        }
+
+        // sorting
+        if (!empty($args['sortBy'])) {
+            $objectList->setOrderKey($args['sortBy']);
+            if (!empty($args['sortOrder'])) {
+                $objectList->setOrder($args['sortOrder']);
+            }
+        }
+
+        // Include unpublished
+//        if (isset($args['published']) && $args['published'] === false) {
+//            $objectList->setUnpublished(true);
+//        }
+
+        /** @var $configuration Configuration */
+        $configuration = $context['configuration'];
+//        $sqlListCondition = $configuration->getSql();
+//
+//        if ($sqlListCondition) {
+//            $conditionParts[] = '(' . $sqlListCondition . ')';
+//        }
+
+        // check permissions
+        $db = Db::get();
+        $tableName = $this->getTypeTableName();
+        $workspacesTableName = 'plugin_datahub_workspaces_' . $this->elementType;
+        $conditionParts[] = ' (
+            (
+                SELECT `read` from ' . $db->quoteIdentifier($workspacesTableName) . '
+                WHERE ' . $db->quoteIdentifier($workspacesTableName) . '.configuration = ' . $db->quote($configuration->getName()) . '
+                AND LOCATE(CONCAT(' . $db->quoteIdentifier($tableName) . '.path,' . $db->quoteIdentifier($tableName) . '.filename),' . $db->quoteIdentifier($workspacesTableName) . '.cpath)=1
+                ORDER BY LENGTH(' . $db->quoteIdentifier($workspacesTableName) . '.cpath) DESC
+                LIMIT 1
+            )=1
+            OR
+            (
+                SELECT `read` from ' . $db->quoteIdentifier($workspacesTableName) . '
+                WHERE ' . $db->quoteIdentifier($workspacesTableName) . '.configuration = ' . $db->quote($configuration->getName()) . '
+                AND LOCATE(' . $db->quoteIdentifier($workspacesTableName) . '.cpath,CONCAT(' . $db->quoteIdentifier($tableName) . '.filename,' . $db->quoteIdentifier($tableName) . '.filename))=1
+                ORDER BY LENGTH(' . $db->quoteIdentifier($workspacesTableName) . '.cpath) DESC
+                LIMIT 1
+            )=1
+        )';
+
+        if (isset($args['filter'])) {
+            $filter = json_decode($args['filter'], false);
+            if (!$filter) {
+                throw new ClientSafeException('unable to decode filter');
+            }
+            $filterCondition = Helper::buildSqlCondition($tableName, $filter);
+            $conditionParts[] = $filterCondition;
+        }
+
+        if ($conditionParts) {
+            $condition = implode(' AND ', $conditionParts);
+            $objectList->setCondition($condition);
+        }
+
+//        $event =  new ListingEvent(
```

Comment:
Should we use the same event or introduce a new one?

Reply:
current one is generic enough I guess !?

Reply:
done